### PR TITLE
Change podNamespacer API

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/30759

Should significantly help for controllers performance.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30774)

### CODE REVIEW

File: pkg/controller/controller_ref_manager.go [hunk 1]
```diff
@@ -53,13 +53,13 @@ func NewPodControllerRefManager(
...
 // labels no longer match the selector.
-func (m *PodControllerRefManager) Classify(pods []api.Pod) (
+func (m *PodControllerRefManager) Classify(pods []*api.Pod) (
 	matchesAndControlled []*api.Pod,
 	matchesNeedsController []*api.Pod,
 	controlledDoesNotMatch []*api.Pod) {
 	for i := range pods {
 		pod := pods[i]
-		if !IsPodActive(pod) {
+		if !IsPodActive(*pod) {
```
Comment (caesarxuchao) [inline pkg/controller/controller_ref_manager.go]: Change IsPodActive to take pointer as well?

**Comment by caesarxuchao** [comment]
Seems you missed some interfaces in tests:
```
pkg/controller/controller_utils_test.go:290: cannot use podList.Items (type []"k8s.io/kubernetes/pkg/api".Pod) as type []*"k8s.io/kubernetes/pkg/api".Pod in argument to FilterActivePods
```

File: pkg/controller/replication/replication_controller.go [hunk 2]
```diff
@@ -660,13 +663,13 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 			return aggregate
 		}
 	} else {
-		podList, err := rm.podStore.Pods(rc.Namespace).List(labels.Set(rc.Spec.Selector).AsSelector())
+		pods, err := rm.podStore.Pods(rc.Namespace).List(labels.Set(rc.Spec.Selector).AsSelector())
```
Comment (caesarxuchao) [inline pkg/controller/replication/replication_controller.go]: This is the code where GC is not enabled, so the problem is there even without GC.

Comment (wojtek-t) [inline pkg/controller/replication/replication_controller.go]: It is - but it seems significantly worse with GC enabled (I think it's related to other loops through those pods).

File: pkg/controller/controller_ref_manager.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/controller/controller_ref_manager.go]: Part of #30768

**Comment by lavalamp** [comment]
This is now part of #30807

**Pull request closed without merging**